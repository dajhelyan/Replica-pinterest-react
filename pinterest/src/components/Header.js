/* eslint-disable jsx-a11y/alt-text */
/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import Navbar from 'react-bootstrap/Navbar'
import Form from 'react-bootstrap/Form'
import Button from 'react-bootstrap/Button'
import Col from 'react-bootstrap/Col'


export const Header = ({ children }) => {
  return (
    <Navbar bg="light" expand="lg" style={{ borderBottom: "groove" }}>
      <img
        src="/images/pinterest.png"
        width="40"
        height="40"
        className="d-inline-block align-top"
        alt="React Bootstrap logo"
      />
      <span style={{marginLeft: "12px"}}><h5>Pinterest</h5></span>
      <Navbar.Collapse id="basic-navbar-nav">
        {
          children
        }
        <Button style={{marginRight: "10px"}} variant="danger">Registrarse</Button>
        <Button variant="secondary">Iniciar sesiÃ³n</Button>
        <Button variant="outline-secondary" style={{marginLeft: "10px", borderRadius: "29px",borderColor: "white"}}><i class="fas fa-ellipsis-h"></i></Button>
      </Navbar.Collapse>
    </Navbar>
  )
}