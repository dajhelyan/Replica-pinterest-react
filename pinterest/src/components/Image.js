import React from 'react';

export const Image = ({ url, alt, forwardref  }) => {
    return (
        <img ref={forwardref} src={url} alt={alt} />

    )

}