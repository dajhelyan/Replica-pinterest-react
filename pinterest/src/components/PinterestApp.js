import React, { useState, useEffect } from 'react';

const id = '4c974f2e37b1799fdb6e91a0f891a25df26b687e9e6eb77816d9988dd5142e59';
const api = `https://api.unsplash.com/photos/?client_id=${id}&per_page=20`;

const PinterestApp = (props) => {

    // trayendo elementos y almacenandolos en una constante
    const initialState = useFetchApi(api)


    return (
        <React.Fragment>
            <Header>
                <Search />
            </Header>
            <p>hola {props.author}</p>

            {/* iterando dentro de componente para agregar un componente de cada imagen por iteracion */}
            <ContainerImg>
                {
                    initialState.map((item) => {
                        return <CardItem key={item.id} {...item} />
                    })
                }
            </ContainerImg>

        </React.Fragment>
    )
}



const Header = ({ children }) => {
    return (
        <header>
            <figure>
                <img src="././public/images/logo.png" alt="logo-pinterest" />
            </figure>
            {children}
            <button>Registrarse</button>
            <button>Iniciar sesi√≥n</button>
        </header>
    )
}

const Search = () => {

    const [query, setQuery] = useState('');

    // const apiSearch = `https://api.unsplash.com/search/photos?&query=${query}&client_id=${id}&per_page=20`


    const searching = () => {
        fetch(`https://api.unsplash.com/search/photos?&query=${query}&client_id=${id}&per_page=20`)
        .then(res => {
            return res.json()
        }).then((result) => {
            console.log(result)
        })
       
    }

    /* generando un evento y controlando con onclick */
    const handleClick = (e) => {
        setQuery(e.target.value);
        console.log(setQuery, 'oii')
    }

    /* const handleSubmit = (e) => {
        e.preventDefault(e)
        useFetchApi(apiSearch)
    } */

    return (
        <div>
            <input type="text" 
                onChange={handleClick}
                placeholder="Buscar"
                value={query}
            />
            <button onClick={searching} >Buscar</button>
        </div>
    )
}




// componente contenedor de imagenes, recibiendo como prop un children(que recibe en el componente app una iteracion con map que retrona un componentecon cada imagen)
const ContainerImg = ({ children }) => {

    return (
        <section>
            <div className="container">
                <div className="card-columns">
                    {children}
                </div>
            </div>
        </section>
    )

}

// componente card por imagen que recibe como prop una todo item(data)
const CardItem = ({ ...item }) => {
    return (
        <div className="card">
            {/* inseratando componente imagen mandando de prop link de cada imagen y descripcion */}
            <Image url={item.urls.raw} alt={item.alt_description} />
        </div>
    )
}

// componente imagen que recibe url de imagen para pintar
const Image = ({ url, alt }) => {
    return (
        <img src={url} alt={alt} height="100" />

    )

}

// construyendo un hook que hace fetch api
const useFetchApi = (api) => {
    const [data, setData] = useState([])

    useEffect(() => {
        // agrengando parametro que trae 20 imagenes por pagina
        fetch(api)
            .then((response) => {
                return response.json();
            })
            .then((data) => {
                setData(data);
            });
    }, [])
    console.log(data, 'popo');
    return data;
}






export default PinterestApp;