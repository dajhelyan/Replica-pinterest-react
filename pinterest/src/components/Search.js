import React, { useRef } from 'react'
import Form from 'react-bootstrap/Form'
import Col from 'react-bootstrap/Col'

export const Search = ({setQuery, query}) => {
    //dando referencia al iut y mediante ele elemneto capturara valor 
    const formRef = useRef();

    const handleClck = () =>  {
        const newQuery = formRef.current.value;
        setQuery(newQuery) 
    }

    return (
        
        <Form inline as={Col} md="9">
            
            <div className="control has-icons-left">
                <input className="input" 
                style={{width: "760px", background: "lightgray"}}
                type="text" 
                ref={formRef}
                defaultValue={query}
                placeholder="Buscar"
                />
                <span className="icon is-small is-left">
                    <i style={{color: "black"}}className="fas fa-search"></i>
                </span>
            </div>
              <button className="button is-success is-light" onClick={handleClck}>Search</button>
        </Form>
    )
}