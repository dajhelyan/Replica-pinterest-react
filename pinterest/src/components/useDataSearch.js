import { useState, useEffect } from 'react'
import Axios from 'axios'

const useDataSearch = (query, pageNum) => {
  const id = '4c974f2e37b1799fdb6e91a0f891a25df26b687e9e6eb77816d9988dd5142e59';

  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(true)
  const [hasMore, setHasMore] = useState(false)

  useEffect(() => {
    Axios(`https://api.unsplash.com/photos/?client_id=${id}&per_page=20`)
      .then(res => {
        return setData(res.data)
      })
  }, []);

  useEffect(() => {
    if (query === '') {
      return;
    }
  }, [])
    useEffect(() => {

      Axios({
        method: "GET",
        url: `https://api.unsplash.com/search/photos?&client_id=${id}&per_page=20`,
        params: { query: query, page: pageNum }
      }).then(res => {
        setLoading(false)
        setHasMore(true)
        setData(res.data.results)
      })


    }, [query, pageNum])
    return { data, loading, hasMore }
  }

export default useDataSearch;