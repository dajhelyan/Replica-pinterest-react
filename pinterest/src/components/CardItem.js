import React from 'react';
import { Image } from './Image'
import Card from 'react-bootstrap/Card';
import Modal from 'react-bootstrap/Modal';
require("react-bootstrap/ModalHeader")

// componente card por imagen que recibe como prop una todo item(data)
export const CardItem = ({ forwardref, ...item }) => {
    const [show, setShow] = React.useState(false);

    return (
        <>
            <Card onClick={() => setShow(true)}>
                <Image forwardref={forwardref} url={item.urls.small} alt={item.alt_description} />
                
            </Card>

            <Modal
                show={show}
                onHide={() => setShow(false)}
                size="sm"
                aria-labelledby="contained-modal-title-vcenter"
                centered
            >
                <Modal.Header closeButton>
                </Modal.Header>
                <Modal.Body>
                    <img src={item.urls.small} alt={item.alt_description} />
                </Modal.Body>
            </Modal>
        </>
    )
}

