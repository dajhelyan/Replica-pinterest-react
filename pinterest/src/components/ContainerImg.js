import React from 'react';
import { CardItem } from '././CardItem'
import CardColumns from 'react-bootstrap/CardColumns'
import Container from 'react-bootstrap/Container'

export const ContainerImg = ({data, forwardRef}) => {

    return (
       
            <CardColumns style={{columnCount: "5", margin: "20px 50px 0px 50px"}} >
                    {
                        data.map((item) => {
                            return <CardItem key={item.id} forwardref={forwardRef} {...item} />
                            
                        })
                    } 
            </CardColumns>
        
    )

}