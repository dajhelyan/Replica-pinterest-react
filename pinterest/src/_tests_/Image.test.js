import React from 'react';
import { shallow } from 'enzyme';
import { Image } from '../components/Image';

// move to setuptest
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'
Enzyme.configure({ adapter: new Adapter() });
// move to setuptest

const fn = jest.fn();

describe('<Image /> snapshot;', () => {
  it('renders with attributes provided', () => {
    const output = shallow(<Image url='url' alt='alt' forwardref={fn} />);
    expect(toJson(output)).toMatchSnapshot();
  });
});