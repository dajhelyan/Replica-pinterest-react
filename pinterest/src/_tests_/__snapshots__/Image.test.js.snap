// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> snapshot; renders with attributes provided 1`] = `
<img
  alt="alt"
  src="url"
/>
`;
